ngi_pipeline/engines/qc_ngi/workflows.py

```diff
@@ -85,14 +85,15 @@ def workflow_fastqc(input_files, output_dir, config):
     fastqc_output_file_tmpls = ("{}_fastqc.zip", "{}_fastqc.html")
     fastq_to_analyze = set()
     for fastq_file in fastq_files:
-        # Get the basename withot extensions (.fastq, .fastq.gz)
         m = re.match(r'([\w-]+).fastq', os.path.basename(fastq_file))
+        #fetch the FCid
+        fc_id = re.search('\d{4,6}_(?P<fcid>[A-Z0-9]{10})', fastq_file).groups()[0]
```

Comment:
isn't this regexp both a bit too specific and unspecific? how about catching on either the entire runfolder name, e.g. `\d{6}_[A-Z0-9-]+_\d{4,}_(?P<fcid>[A-Z0-9]{10})` or relying on the organization and get the fcid from the parent folder, e.g. `os.path.dirname(fastq_file).split("_")[-1]`?